noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_init.nr

```diff
@@ -20,63 +23,23 @@ struct PrivateKernelInitCircuitPrivateInputs {
 }
 
 impl PrivateKernelInitCircuitPrivateInputs {
-    // Confirm that the TxRequest (user's intent) 
-    // matches the private call being executed
-    fn validate_this_private_call_against_tx_request(self) {
-        let tx_request = self.tx_request;
-        // Call stack item for the initial call
-        let call_stack_item = self.private_call.call_stack_item;
-
-        // Checks to ensure that the user's intent matches the initial private call
-        //
-        // We use the word correct to denote whether it matches the user intent.
-        //
-        // Ensure we are calling the correct initial contract
-        let origin_address_matches = tx_request.origin.eq(call_stack_item.contract_address);
-        assert(origin_address_matches, "origin address does not match call stack items contract address");
-        //
-        // Ensure we are calling the correct initial function in the contract
-        let entry_point_function_matches = tx_request.function_data.hash() == call_stack_item.function_data.hash();
-        assert(
-            entry_point_function_matches, "tx_request function_data must match call_stack_item function_data"
-        );
-        //
-        // Ensure we are passing the correct arguments to the function.
-        let args_match = tx_request.args_hash == call_stack_item.public_inputs.args_hash;
-        assert(args_match, "noir function args passed to tx_request must match args in the call_stack_item");
-        //
-        // Ensure we are passing the correct tx context
-        let tx_context_matches = tx_request.tx_context == call_stack_item.public_inputs.tx_context;
-        assert(tx_context_matches, "tx_context in tx_request must match tx_context in call_stack_item");
-    }
-
     fn validate_inputs(self) {
-        let call_stack_item = self.private_call.call_stack_item;
-
-        let function_data = call_stack_item.function_data;
-        assert(function_data.is_private, "Private kernel circuit can only execute a private function");
```

Comment:
Removed this check since `validate_contract_address` verifies the function against the `private_functions_root`.